base/number.jl

```diff
@@ -63,4 +63,6 @@ zero{T<:Number}(::Type{T}) = convert(T,0)
 one(x::Number)  = oftype(x,1)
 one{T<:Number}(::Type{T}) = convert(T,1)
 
+_default_type(::Type{Number}) = Int
```

Comment:
This was missing for preserving containers of `Number` as `eltype`. When the container is of an abstract type, `promote_op` uses `_default_type` and then the type found is '`typejoin`ed with the `eltype` of the container. I guess I could use `Float64`, but I thought that `Int` should give a reasonable result (that is, not `Any`) generally.